File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cs
public class Solution {
    public int FindMin(int[] nums) {
        int n = nums.Length;

        int l = 0, r = n - 1;
        while (l < r) {
            int m = l + (r - l) / 2;
            if (nums[m] < nums[r]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return nums[l];
    }
}

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0, right = nums.size();

        int pivot = INT_MAX;
        while(left < right)
        {
            int mid = left + (right - left) / 2;
            
            
            if(nums[mid] > nums[left])
            {
                pivot = min(pivot, nums[left]);
                left = mid + 1;
            }
            else
            {
                pivot = min(pivot, nums[mid]);
                right = mid;
            }
        }

        return pivot;
    }
};